wasm/jsapi/module/moduleSource.any.js

```diff
@@ -0,0 +1,37 @@
+// META: global=window,dedicatedworker,jsshell,shadowrealm
+// META: script=/wasm/jsapi/wasm-module-builder.js
+// META: script=/wasm/jsapi/assertions.js
+
+let emptyModuleBinary;
+setup(() => {
+  emptyModuleBinary = new WasmModuleBuilder().toBuffer();
+});
+
+test(() => {
+  assert_equals(typeof AbstractModuleSource, "undefined");
+  const AbstractModuleSource = Object.getPrototypeOf(WebAssembly.Module).constructor;
+  assert_equals(AbstractModuleSource.name, "AbstractModuleSource");
+  assert_not_equals(AbstractModuleSource, Function);
+}, "AbstractModuleSource intrinsic");
+
+test(() => {
+  const AbstractModuleSourceProto = Object.getPrototypeOf(WebAssembly.Module.prototype).constructor;
```

Comment:
`WebAssembly.Module.prototype` is a plain object, this should probably be
```suggestion
  const AbstractModuleSourceProto = Object.getPrototypeOf(WebAssembly.Module).prototype;
```

but I'm not sure about what this test is testing actually.

Reply:
Based on the feedback in the previous comment I believe this should be:

```suggestion
  const AbstractModuleSourceProto = Object.getPrototypeOf(WebAssembly.Module.prototype);
```

where what is being tested is that `WebAssembly.Module.prototype` extends from the abstract module source prototype, just as WebAssembly.Module extends from abstract module source.